src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLAlertImpl.java

```diff
@@ -190,38 +195,128 @@
     return environment -> getSource(environment).getEntities().stream()
             .map(entitySelector -> {
               if (entitySelector instanceof EntitySelector.Stop) {
-                return getRoutingService(environment).getStopForId(((EntitySelector.Stop) entitySelector).stopId);
+                FeedScopedId id = ((EntitySelector.Stop) entitySelector).stopId;
+                Stop stop = getRoutingService(environment).getStopForId(id);
+                return getAlertEntityOrUnknown(stop, id.toString(), "stop");
               }
               if (entitySelector instanceof EntitySelector.Agency) {
-                return getRoutingService(environment).getAgencyForId(((EntitySelector.Agency) entitySelector).agencyId);
+                FeedScopedId id = ((EntitySelector.Agency) entitySelector).agencyId;
+                Agency agency = getRoutingService(environment).getAgencyForId(id);
+                return getAlertEntityOrUnknown(agency, id.toString(), "agency");
               }
               if (entitySelector instanceof EntitySelector.Route) {
-                return getRoutingService(environment).getRouteForId(((EntitySelector.Route) entitySelector).routeId);
+                FeedScopedId id = ((EntitySelector.Route) entitySelector).routeId;
+                Route route = getRoutingService(environment).getRouteForId(id);
+                return getAlertEntityOrUnknown(route, id.toString(), "route");
               }
               if (entitySelector instanceof EntitySelector.Trip) {
-                return getRoutingService(environment).getTripForId().get(((EntitySelector.Trip) entitySelector).tripId);
+                FeedScopedId id = ((EntitySelector.Trip) entitySelector).tripId;
+                Trip trip = getRoutingService(environment).getTripForId().get(id);
+                return getAlertEntityOrUnknown(trip, id.toString(), "trip");
               }
               if (entitySelector instanceof EntitySelector.TripPattern) {
-                return getRoutingService(environment).getTripPatternForId(((EntitySelector.TripPattern) entitySelector).tripPatternId);
+                FeedScopedId id = ((EntitySelector.TripPattern) entitySelector).tripPatternId;
+                TripPattern tripPattern = getRoutingService(environment).getTripPatternForId(id);
+                return getAlertEntityOrUnknown(tripPattern, id.toString(), "trip pattern");
               }
               if (entitySelector instanceof EntitySelector.StopAndRoute) {
```

Comment:
We could also validate that the stop is actually on the route/trip but I think it's a valid use case that you create an alert "Route x no longer stops at stop y" and want to display that on route, for example.

Reply:
Decided that alerts should have the original XandY entity type even though they are not really linked together (anymore).